src/2dplane/GridStateSpace.cpp

```diff
@@ -16,6 +15,30 @@ bool GridStateSpace::stateValid(const Vector2f &pt) const {
     return PlaneStateSpace::stateValid(pt) && !_obstacleGrid.obstacleAt(_obstacleGrid.gridSquareForLocation(pt));
 }
 
+Vector2f GridStateSpace::intermediateState(const Vector2f &source, const Vector2f &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const {
+    float stepSize = prevStepSize;
+    Vector2f delta = target - source;
+    delta = delta / delta.norm();   //  unit vector
+    if (prevStepSize == 0) {
+        stepSize = defaultStepSize;
+    } else {
+        float dist = _obstacleGrid.nearestObstacleDist(source);
+        if (dist > prevStepSize * distScale()) { //grows if dist > max tolerable dist
+            stepSize = prevStepSize * ascGrowthRate;
+        } else if (dist < prevStepSize * distScale()) { //shrinks if dist < max tolerable dist
+            stepSize = prevStepSize / (ascGrowthRate * 2);
+        }
+        // stepSize = stepSize * pow(dist, 0.25); // this is an alternative to the if block that is more elegant but more computationally intensive because exponents
+    }
+
+    if (stepSize > maxStepSize() * defaultStepSize) { 
+        //sets max limit for stepsize
+        stepSize = maxStepSize() * defaultStepSize;
```

Comment:
It seems a little unintuitive that there's a property called `maxStepSize`, but it's not actually the max.